tests/fixtures/valid_analysis/data_models/example_data_model_python.yml

```diff
@@ -1,7 +1,7 @@
 AnalysisType: datamodel
 DataModelID: GSuite.Events.DataModel
 Enabled: true
-Filename: example_data_model_python.py
+Filename: GSuite.Events.DataModel.py
```

Comment:
curious, why the name change?

Reply:
I agree, unless there is a specific reason I'd prefer to keep extraneous `.`'s out of filenames.

Reply:
+1 bump

Reply:
Unfortunately there is an important reason for renaming. The `DataModel` constructor fails if it cannot create a module and uses a specific naming convention for the temporary file it stores. The reason for the failure is that the fake filesystem does not patch the `importlib` loaders, because these functions actually operate on a VM level. I assume it would actually be insecure if that was allowed as someone could shadow a known package dynamically. Thus, we need to rename the file to actually exist in the filesystem as the `DataModel` expects.

I expect that it may need some a few more tweaks until we can fully align both codebases, so we could fix this in a subsequent PR, no problem with that. 

Note that it would be an improvement if the constructor performed no operations, but instead just initialized attributes. Why? Because if the constructor calls instance functions, especially those with side-effects or external dependencies (e.g. filesystem), that could fail, it becomes inherently difficult to test, because you cannot make reliable assertions on an object that cannot be constructed. The pattern that is followed in such cases is to use a property getter and check if a protected/private attribute is already populated:

```python
@property
def module(self):
   if self._module is None:
     self._module = self._get_module()
   return self._module
```